### PR TITLE
feat: implemented the search users by first name

### PR DESCRIPTION
**Pull request by StarryFolf**

### CODE REVIEW

File: src/Infrastructure/Repositories/UserRepository.cs [hunk 1]
```diff
@@ -40,4 +40,13 @@ public async Task<User> CreateUserAsync(User user)
         await _connection.ExecuteAsync(sql, queryArguments, transaction: _transaction);
         return user;
     }
+
+    public async Task<IEnumerable<User>> GetUserByNameAsync(String firstName)
+    {
+        var sql = @"SELECT Username, Email, FirstName, LastName, DepartmentId, Role, Position, IsActive, IsActivated" +
+                  "FROM Users" +
```
Comment (ChienNQuang) [inline src/Infrastructure/Repositories/UserRepository.cs:47]: add whitespace at the end

File: src/Infrastructure/Repositories/UserRepository.cs [hunk 2]
```diff
@@ -40,4 +40,13 @@ public async Task<User> CreateUserAsync(User user)
         await _connection.ExecuteAsync(sql, queryArguments, transaction: _transaction);
         return user;
     }
+
+    public async Task<IEnumerable<User>> GetUserByNameAsync(String firstName)
+    {
+        var sql = @"SELECT Username, Email, FirstName, LastName, DepartmentId, Role, Position, IsActive, IsActivated" +
```
Comment (ChienNQuang) [inline src/Infrastructure/Repositories/UserRepository.cs:46]: add whitespace at the end

**Review by ChienNQuang — CHANGES_REQUESTED** [review]
Could have had pagination but ok for now, some adjustments need to be made

File: src/Infrastructure/Repositories/UserRepository.cs (hunk 2 above)
Comment (StarryFolf) [reply src/Infrastructure/Repositories/UserRepository.cs:46]: thanks, didnt notice that

**Review by Vzart — APPROVED** [review]
Thank you

**Review by kaitoz11 — APPROVED** [review]

**Pull request merged**